src/lib.rs

```diff
@@ -26,10 +26,9 @@ extern crate mio;
 #[cfg(all(not(feature = "force-inprocess"), any(target_os = "linux",
                                                 target_os = "freebsd")))]
 extern crate fnv;
-#[cfg(all(feature = "memfd", not(feature = "force-inprocess"),
-          target_os="linux"))]
-#[macro_use]
-extern crate syscall;
+#[cfg(all(not(feature = "force-inprocess"), any(target_os = "linux",
```

Comment:
Related to myfreeweb/shmemfdrs#1. The old implementation had a default that was used if the target os was not linux (with memfd) or freebsd.

Reply:
The issue is no longer valid, but this still does not allow e.g. openbsd to use the `shmemfdrs` crate

Reply:
You mean the hardcoded `any(target_os="linux", target_os="freebsd")` in here? Yeah, I would also like to replace them with… `all(unix, not(any(target_os="macos", target_os="ios")))` I guess?

Reply:
Actually, I think you are right again here. I seem to have done the same thing with `extern crate mio` and this is the same as the `cfg` for `mod unix`, so there should probably be a separate issue to possibly modify the `cfg`s.

Reply:
This is a bit confusing, since the implementation of the `unix` back-end should work with any Unix-like OS, but the conditional *selecting* the `unix` back-end (in https://github.com/servo/ipc-channel/blob/master/src/platform/mod.rs ) only works on specific platforms :-(

The only real change here as far as I can tell is that the `shmemfdrs` dependency requires introducing (yet another) redundant copy of the `unix` conditional.

(A long-standing issue on my personal ToDo-list is to try to factor out the platform conditionals in a central place, so everything else would only have to test for a custom `platform_unix` conditional, or something along these lines...)

Reply:
Yeah. Arguably, it's not *another* copy, it's `syscall`'s copy :D

+1 that it's a separate issue

Reply:
@dlrobertson oh, I see you already figured that out :-)

Reply:
@myfreeweb nah, `syscall` doesn't have an exact copy of the `unix` conditional, since that one is (and always was) Linux-specific, so it didn't have the `not generic enough` problem. But yeah, separate issue.